packages/cli/src/commands/upgrade.js

```diff
@@ -239,3 +249,34 @@ async function refreshPrismaClient(task, { verbose }) {
     console.log(c.error(e.message))
   }
 }
+
+const dedupeDeps = async (task, { verbose }) => {
+  try {
+    // Get current yarn version
+    const { stdout } = await execa('yarn', ['--version'], {
+      cwd: getPaths().base,
+    })
+    if (!SEMVER_REGEX.test(stdout)) {
+      throw new Error('Unable to verify yarn version.')
+    }
+
+    // Run dedupe if yarn is version <=1.x
+    const version = stdout.match(SEMVER_REGEX)[0]
+    const majorVersion = parseInt(version.split('.')[0])
+    if (majorVersion > 1) {
```

Comment:
Thanks for looking into it @simoncrypta! I didn't opted for this because:
- in edge cases, for example if `yarn` is not present, the `stdout` could be anything and if it starts with "1", we would mistake it for compatibility.
- Versions like 10.0.0 - Not happening anytime soon 😅 but still.. 😬

Reply:
Make sense! Thank you for the explanation 😊